Phonebook.Frontend/src/app/shared/interceptors/HttpRedirectToLogin.ts

```diff
@@ -0,0 +1,31 @@
+import { HttpEvent, HttpHandler, HttpInterceptor, HttpRequest, HttpResponse, HttpErrorResponse, HttpHeaders, HttpClient } from '@angular/common/http';
+import { Injectable } from '@angular/core';
+import { Observable, empty} from 'rxjs';
+import { map, catchError } from 'rxjs/operators';
+
+@Injectable({ providedIn: 'root' })
+export class HttpRedirectToLogin implements HttpInterceptor {
+  constructor(private http: HttpClient) { }
+
+  public intercept(req: HttpRequest<any>, next: HttpHandler): Observable<HttpEvent<any>> {
+    return next.handle(req).pipe(
+      map(d => {
+        if (d instanceof HttpResponse && d.status == 401 && d.headers.has("Location")) {
```

Comment:
Evtl. sollten wir das noch etwas einschränkender machen und nur für die Domain auf der das Backend läuft ausführen

Reply:
das geht nicht, da durch unseren Proxy alles die selbe Domain hat :D

Reply:
Ich guck dann nochmal rein. Das ist mir im Moment klar, aber vielleicht haben wir in zukunft noch andere domains z.B. den bilderservice.